content/docs/i18n.mdx

```diff
@@ -0,0 +1,243 @@
+
+### `defaultMessage`
+
+The default message is provided by JSONForms and can act as a fallback or could be translated.
+
+:::note
+
+If the defaultMessage is `undefined` it should not be modified and still return as `undefined` (and not an empty string or similar).
+JSON Forms will use `undefined  when the message could be skipped or another generic key could be tried. 
+
+:::
+
+### `values`
+
+The values can contain additional context, if the key of the string is not enough to determine the correct translation.
+For example all error translations will have the error as part of the values object.
+
+## Error Customizations
+
+TODO: Description
+
+#### `<i18nkey>.error.custom`
+
+If `error.custom` is set, just a single error will be output, no matter how many errors are present.
+The `translate` function is called for `error.custom`
+The `translateError` function is called for each error.
+By default it behaves like this:
+
+## `translateError`
+
+The `translateError` is called for every AJV error. 
+`translateError` is an optional parameter to the i18n object. Usually it is not expected to customize this option.
+By default error it works like this:
+
+`<i18nkey>` in the sections below refers to the key of the field (see `key` section above).
+
+### Evaluation order
+
+#### `<i18nkey>.error.<keyword>`
+
+e.g. `name.error.required`
+
+If no `error.custom` is set, the translator will look for a concrete message for a specific field and a specific error. 
+In the example above, the function will look for a `required` error on the `name` field.
+
+#### `error.<keyword>`
+
+e.g. `error.required`
+
+If no specific field error message is set, the translator will then look for an error message, that is independent of the field. 
+In the example case above, the function will look for a `required` error translation.
+
+#### Default error translator
+
+Next the translator will look for a default message, which is provided by the core framework.
+
+#### Default AJV error message
+
+If none of the above are set and no default error is available, the error message provided by AJV will be output.
+
+### Example
+
+If we have the following schema:
+```js
+{
+  phone: {
+    type: "string",
+    minLength: 10,
+    pattern: "^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$"
+  }
+}
+```
+And the order in which the error keys are evaluated is the following:
+- `phone.error.custom`: if this is set, the `pattern` and `minLength` error will be ignore and just this message is output
+- `phone.error.pattern` & `phone.error.minLength`
+- `error.pattern` & `error.minLength`
+
+## Enum translation
+
+Enum translations are a little different then the other control translations. 
+In order to translate enum values, you need to add one additional attribute to the translation for every value. 
+For example:
+Let's assume we have a field `Gender`, which has three possible values: 
+- `Male`
+- `Female`
+- `Other`
```

Comment:
Let's assume we have an enum attribute `gender`, which looks like this:

```js
{
  gender: {
    type: "string",
    enum: ["male", "female", "other"]
  }
}
```

In this case the `translate` function will be invoked with the keys `gender.male`, `gender.female` and `gender.other` to translate these enum values.
In case `gender` had an `i18n` property, it would be used instead, i.e. `<i18nkey>.male` etc.

Let's assume we have a `oneOf` enum attribute gender which looks like this:
```js
{
  gender: {
    oneOf: [
      {
        title: "Male",
        const: 0
      },
      {
        title: "Female",
        const: "f",
        i18n: "fem"
      },
      {
        const: null
      }
    ]
  }
}
```
Here the requested keys are:
 - `gender.Male` - property path + `title` of the `oneOf` entry
 - `fem` - direct usage of the `i18n` property for the `oneOf` entry
 - `null` - the `title` attribute is missing, therefore the `null` value is stringified to `'null'`.